src/lib/plot/bar.typ

```diff
@@ -52,19 +52,30 @@
     let right = left + w
     let width = (right - left) / len
 
+
     if self.mode in ("basic", "clustered") {
       left = left + width * n
       right = left + width
     }
-
     if (left <= x-axis.max and right >= x-axis.min and
         y-min <= y-axis.max and y-max >= y-axis.min) {
       left = calc.max(left, x-axis.min)
       right = calc.min(right, x-axis.max)
       y-min = calc.max(y-min, y-axis.min)
       y-max = calc.min(y-max, y-axis.max)
 
-      draw.rect((left, y-min), (right, y-max))
+
+      draw.group(ctx => {
+        draw.rect((left, y-min), (right, y-max))
+        if ctx.style.data-label != none {
+          let offset = ctx.style.data-label.at("offset")
+          let size = ctx.style.data-label.at("text-size")
+          let data_label = text(size:size)[#y-max]
```

Comment:
The label itself should come from a callback, so the user can format it etc.
To do this, add a `data-label` or `bar-label` (?) argument to `add-bar`.
We could allow some special values like `"max"` to auto use a callback that shows the maximum value.

Such callback should receive the "current" bar as argument.

I do not like having a font-size as style attribute. I prefer the user setting the size via the label callback.